gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -456,13 +457,40 @@ public WriteChannel writer(BlobWriteOption... options) {
    * Generates a signed URL for this blob. If you want to allow access to for a fixed amount of time
    * for this blob, you can use this method to generate a URL that is only valid within a certain
    * time period. This is particularly useful if you don't want publicly accessible blobs, but don't
-   * want to require users to explicitly log in.
+   * want to require users to explicitly log in. Signing a URL requires a service account
+   * and its associated key. If a {@link AuthCredentials.ServiceAccountAuthCredentials} was passed
+   * to {@link StorageOptions.Builder#authCredentials(AuthCredentials)} or the default credentials
+   * are being used and the environment variable {@code GOOGLE_APPLICATION_CREDENTIALS} is set, then
+   * {@code signUrl} will use that service account and associated key to sign the URL. If this
+   * is not the case, a service account with associated key can be passed to {@code signUrl} using
```

Comment:
I know but I am trying to make clear the point here is having a "key". While "service account credentials" is used in official docs it's also a vague term, a user might assume that since he is logged in gcloud he already has the "credentials" he needs. I would stick to the verbose "a service account with associated key" if you don't feel too strong about this.